### PR TITLE
[Merged by Bors] - feat(Algebra/Polynomial/Bivariate): add lemmas for bivariate polynomials

### PR DESCRIPTION
**Pull request by Multramate**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

**Comment by Multramate** [comment]
These are not strictly necessary, because we can just rewrite `evalEval` into `eval` (or into `evalEvalRingHom`), and apply the corresponding lemmas of `eval` (or of `evalEvalRingHom`), but are API that are "nice to have".

I noted, when doing this, that there are a few missing gaps in the API of `eval`, but I'll not touch them here.

File: Mathlib/Algebra/Polynomial/Bivariate.lean [hunk 1]
```diff
@@ -41,15 +41,88 @@ abbrev CC (r : R) : R[X][Y] := C (C r)
 lemma evalEval_C (x y : R) (p : R[X]) : (C p).evalEval x y = p.eval x := by
   rw [evalEval, eval_C]
 
+lemma evalEval_CC (x y : R) (p : R) : (CC p).evalEval x y = p := by
+  rw [evalEval_C, eval_C]
+
+lemma evalEval_zero (x y : R) : (0 : R[X][Y]).evalEval x y = 0 := by
+  simp only [evalEval, eval_zero]
+
+lemma evalEval_one (x y : R) : (1 : R[X][Y]).evalEval x y = 1 := by
```
Comment (semorrison) [inline Mathlib/Algebra/Polynomial/Bivariate.lean:50]: I am surprised that most of the lemmas here are not `@[simp]`!

Comment (Multramate) [reply Mathlib/Algebra/Polynomial/Bivariate.lean:50]: I guess they should be...

File: Mathlib/Algebra/Polynomial/Bivariate.lean [hunk 2]
```diff
@@ -41,15 +41,101 @@ abbrev CC (r : R) : R[X][Y] := C (C r)
...
+@[simp]
+lemma evalEval_natCast (x y : R) (n : ℕ) : (n : R[X][Y]).evalEval x y = n := by
+  simp only [evalEval, eval_natCast]
+
+@[simp]
 lemma evalEval_X (x y : R) : X.evalEval x y = y := by
   rw [evalEval, eval_X, eval_C]
 
+@[simp]
+lemma evalEval_add (x y : R) (p q : R[X][Y]) :
```
Comment (semorrison) [inline Mathlib/Algebra/Polynomial/Bivariate.lean:70]: Ugh, this is not a simp lemma for MvPolynomial, but is for Polynomial. Given that inconsistency, I don't mind what happens here, but if you feel like looking into that later, that would be wonderful. :-)

File: Mathlib/Algebra/Polynomial/Bivariate.lean [hunk 3]
```diff
@@ -41,15 +41,101 @@ abbrev CC (r : R) : R[X][Y] := C (C r)
...
+lemma evalEval_zero (x y : R) : (0 : R[X][Y]).evalEval x y = 0 := by
+  simp only [evalEval, eval_zero]
+
+@[simp]
+lemma evalEval_one (x y : R) : (1 : R[X][Y]).evalEval x y = 1 := by
+  simp only [evalEval, eval_one]
+
+@[simp]
+lemma evalEval_ofNat (x y : R) (n : ℕ) [n.AtLeastTwo] :
+    (OfNat.ofNat n : R[X][Y]).evalEval x y = OfNat.ofNat n := by
```
Comment (semorrison) [inline Mathlib/Algebra/Polynomial/Bivariate.lean:58]: Is it necessary to write OfNat.ofNat here, rather than just n?

Comment (Ruben-VandeVelde) [inline Mathlib/Algebra/Polynomial/Bivariate.lean:58]: Does this work without no_index?

Comment (Multramate) [reply Mathlib/Algebra/Polynomial/Bivariate.lean:58]: What is `no_index`?

Comment (Multramate) [reply Mathlib/Algebra/Polynomial/Bivariate.lean]: I copied the corresponding lemma for `eval`. Without the `Ofnat.ofNat` this is the same as `evalEval_natCast`.

**Review by jcommelin — COMMENTED** [review]
Thanks :tada:

bors merge

**Pull request closed without merging**